code/modules/mob/living/silicon/robot/robot.dm

```diff
@@ -535,6 +535,19 @@
 			toner = tonermax
 			qdel(W)
 			to_chat(user, "<span class='notice'>You fill the toner level of [src] to its max capacity.</span>")
+
+	else if(istype(W, /obj/item/device/assembly/flash))
+		if(!opened)
+			to_chat(user, "<span class='warning'>You need to open the panel to repair the headlamp!</span>")
```

Comment:
I think these two lines need to be removed and 539 needs to become `else if(istype(W, /obj/item/device/assembly/flash) && opened)` otherwise you'll break flashing borgs